### ISSUE TITLE
Query without vectors similar search

### ISSUE THREAD

**Issue by yhmo** [milvus-io/milvus#4401]
**Please state your issue using the following template and, most importantly, in English.**
Sometimes user just want to filter out some entities, no need to do similar search for vectors. 

**Is your feature request related to a problem? Please describe.**
This feature would be very useful when Milvus is used for both storing and retrieving vectors. Where the stored vectors might be needed for features other than only similarity search.

**Comment by yhmo** [comment]
Consider this feature need to add new sdk interface, set the milestone to major version 0.12.0.

**Comment by yhmo** [comment]
Add a new api filter() for each sdk, the usage is like:

```
dsl = {
     "bool": {
         "must":[
             {
                 "term": {"A": [1, 2]}
             }
         ]
     }
}
id_array = milvus.filter(collection_name, dsl)
required_fields = ["A", "B"]
for id in id_array:
    ids = [id]
    entity = milvus.get_entity_by_id(collection_name, ids, required_fields)
    entity = entity.dict()
    print("entity = ", res)
```

**Comment by tailtq** [comment]
Agree, please consider it!

**Comment by keotic** [comment]
@yhmo will this filter api will also support filtering ANN search vectors by fields, for example, return top k similar vectors that match the dsl condition?

**Comment by yhmo** [comment]
This feature has been implemented in Milvus v2.0.
Please refer the python example: https://github.com/milvus-io/pymilvus/blob/master/examples/query.py

@keotic 
I think your requirement is like #6911. We add a new api named "calc_distance()" for this feature. The example: https://github.com/milvus-io/pymilvus/blob/master/examples/example_distance.py

**Comment by keotic** [comment]
@yhmo our requirement is a bit different than  #6911, while "calc_distance()" supports vectors ids, it calculates distance between sets.
Our request is for supporting query() and dsl where the query payload is entities ids  (instead of vectors) and the result contains distances and entities ids ( not distances and vectors )

it seem that the "query()" API supports our request, by using the "expr" and  "output_fields" parameters, are we correct?
